File: editor/en/[325]Maximum Size Subarray Sum Equals k.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxSubArrayLen = function (nums, k) {
    const map = new Map();
    let max = 0;
    let sum = 0;

    map.set(0, -1);

    for (let i = 0; i < nums.length; i++) {
        sum += nums[i];

        if (map.has(sum - k)) {
            max = Math.max(max, i - map.get(sum - k));
        }

        if (!map.has(sum)) {
            map.set(sum, i);
        }
    }

    return max;
};
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[325]Maximum Size Subarray Sum Equals k.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int maxSubArrayLen(int[] nums, int k) {
        int sum = 0;
        int max = 0;
        Map<Integer, Integer> map = new HashMap<>();

        map.put(0, -1);

        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];

            if (map.containsKey(sum - k)) {
                max = Math.max(max, i - map.get(sum - k));
            }

            if (!map.containsKey(sum)) {
                map.put(sum, i);
            }
        }

        return max;
    }
}
//leetcode submit region end(Prohibit modification and deletion)
